src/main/java/hudson/plugins/jira/CredentialsHelper.java

```diff
@@ -0,0 +1,86 @@
+package hudson.plugins.jira;
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
+
+import com.cloudbees.plugins.credentials.CredentialsMatchers;
+import com.cloudbees.plugins.credentials.CredentialsProvider;
+import com.cloudbees.plugins.credentials.CredentialsScope;
+import com.cloudbees.plugins.credentials.SystemCredentialsProvider;
+import com.cloudbees.plugins.credentials.common.StandardUsernamePasswordCredentials;
+import com.cloudbees.plugins.credentials.domains.URIRequirementBuilder;
+import com.cloudbees.plugins.credentials.impl.UsernamePasswordCredentialsImpl;
+import hudson.security.ACL;
+import hudson.util.Secret;
+import jenkins.model.Jenkins;
+import org.apache.commons.lang.StringUtils;
+
+/**
+ * Helper class for vary credentials operations.
+ *
+ * @author Zhenlei Huang
+ */
+public class CredentialsHelper {
+	private static final Logger LOGGER = Logger.getLogger(CredentialsHelper.class.getName());
+
+	@CheckForNull
+	public static StandardUsernamePasswordCredentials lookupSystemCredentials(@CheckForNull String credentialsId, @CheckForNull URL url) {
+		if (credentialsId == null) {
+			return null;
+		}
+		return CredentialsMatchers.firstOrNull(
+				CredentialsProvider.lookupCredentials(
+						StandardUsernamePasswordCredentials.class,
+						Jenkins.getInstance(),
+						ACL.SYSTEM,
+						URIRequirementBuilder.fromUri(url != null ? url.toExternalForm() : null).build()
+				),
+				CredentialsMatchers.withId(credentialsId)
+		);
+	}
+
+	public static StandardUsernamePasswordCredentials migrateCredentials(@Nonnull String username, String password, @CheckForNull URL url) {
+		StandardUsernamePasswordCredentials cred = null;
+
+		List<StandardUsernamePasswordCredentials> credentials = CredentialsMatchers.filter(
+				CredentialsProvider.lookupCredentials(
+						StandardUsernamePasswordCredentials.class,
+						Jenkins.getInstance(),
+						ACL.SYSTEM,
+						URIRequirementBuilder.fromUri(url != null ? url.toExternalForm() : null).build()
+				),
+				CredentialsMatchers.withUsername(username)
+		);
+		for (StandardUsernamePasswordCredentials c : credentials) {
+			if (StringUtils.equals(password, Secret.toString(c.getPassword()))) {
+				cred = c;
+				break; // found
```

Comment:
we could return `c` here and drop null check below

Reply:
+1